scrypto-tests/tests/address.rs

```diff
@@ -0,0 +1,436 @@
+use scrypto::address::{
+    AddressError, Bech32Decoder, Bech32Encoder, ACCOUNT_COMPONENT_ADDRESS_ENTITY_ID,
+    COMPONENT_ADDRESS_ENTITY_ID, PACKAGE_ADDRESS_ENTITY_ID, RESOURCE_ADDRESS_ENTITY_ID,
+    SYSTEM_COMPONENT_ADDRESS_ENTITY_ID,
+};
+use scrypto::core::Network;
+use scrypto::prelude::{ComponentAddress, PackageAddress, ResourceAddress};
+
+use bech32::{self, ToBase32, Variant};
+
+fn generate_u8_array(entity_byte: u8) -> [u8; 27] {
+    [
+        entity_byte,
+        0,
+        0,
+        0,
+        0,
+        0,
+        0,
+        0,
+        0,
+        0,
+        0,
+        0,
+        0,
+        0,
+        0,
+        0,
+        0,
+        0,
+        0,
+        0,
+        0,
+        0,
+        0,
+        0,
+        0,
+        0,
+        0,
+    ]
+}
```

Comment:
I've added the `address` macro but kept this method here since the macro creates an address with a correct entity byte and here I'm testing for addresses with mismatch entity bytes :)